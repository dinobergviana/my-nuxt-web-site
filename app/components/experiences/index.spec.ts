import { mount } from "@vue/test-utils";
import { describe, it, expect, beforeEach } from "vitest";
import Experiencias from "./index.vue";

describe("Seção Experiencias", () => {
  let wrapper: ReturnType<typeof mount>;

  beforeEach(() => {
    wrapper = mount(Experiencias);
  });

  it("renderiza corretamente o componente", () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('exibe o título principal "experiências"', () => {
    const title = wrapper.find("h2");
    expect(title.exists()).toBe(true);
    expect(title.text()).toBe("experiências");
  });

  it("renderiza todas as experiências profissionais", () => {
    const companies = ["Enjoei", "Educacros", "Warren Investimentos", "BoxTi"];
    companies.forEach((nome) => {
      expect(wrapper.text()).toContain(nome);
    });
  });

  it("exibe cargos e períodos de cada experiência", () => {
    const expectedTexts = [
      "Desenvolvedor Frontend",
      "Desenvolvedor",
      "Mai 2024 - Mai 2025",
      "Set 2022 - Abr 2024",
      "Jan 2022 - Ago 2022",
      "Mar 2021- Dez 2021",
    ];

    expectedTexts.forEach((texto) => {
      expect(wrapper.text()).toContain(texto);
    });
  });

  it("exibe as tecnologias usadas em cada experiência", () => {
    const tecnologies = [
      "Vue.js",
      "Vuex",
      "Typescript",
      "Bootstrap Vue",
      "Node.js",
      "PHP",
      "MySQL",
    ];
    tecnologies.forEach((tech) => {
      expect(wrapper.text()).toContain(tech);
    });
  });

  it("mantém a estrutura HTML e classes conforme esperado (snapshot)", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("possui classes principais de layout", () => {
    const section = wrapper.find("section");
    expect(section.classes()).toContain("grid");
    expect(section.classes()).toContain("min-h-screen");
  });

  it("usa classes de modo escuro corretamente", () => {
    const elementsWithDarkMode = wrapper.findAll('[class*="dark:"]');
    expect(elementsWithDarkMode.length).toBeGreaterThan(0);
  });
});
