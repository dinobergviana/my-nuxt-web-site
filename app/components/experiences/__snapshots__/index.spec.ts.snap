// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Seção Experiencias > mantém a estrutura HTML e classes conforme esperado (snapshot) 1`] = `
"<section class="min-h-screen grid border-t border-gray-300 dark:border-t dark:border-gray-700 md:grid-cols-2 py-8 px-4 text-gray-700 dark:bg-gray-800">
  <div class="flex items-center justify-center mx-auto py-[80px]">
    <h2 class="dark:text-white text-5xl md:text-5xl font-bold m-0 px-4"> experiências </h2>
  </div>
  <div class="dark:text-white flex flex-col justify-center gap-4 p-4">
    <div>
      <h3 class="font-bold text-lg">Enjoei</h3>
      <p class="text-gray-700 font-semibold dark:text-white"> Desenvolvedor Frontend </p>
      <p class="dark:text-gray-300">Mai 2024 - Mai 2025</p><span class="dark:text-gray-300">Trabalhei com:</span><i class="dark:text-gray-300"> Vue.js</i><i class="dark:text-gray-300">, Vuex</i><i class="dark:text-gray-300">, Typescript</i><i class="dark:text-gray-300">, RailsVitest</i><i class="dark:text-gray-300">, Vue Test Utils</i>
    </div>
    <div>
      <h3 class="font-bold text-lg">Educacros</h3>
      <p class="text-gray-700 font-semibold dark:text-white"> Desenvolvedor Frontend </p>
      <p class="dark:text-gray-300">Set 2022 - Abr 2024</p><span class="dark:text-gray-300">Trabalhei com:</span><i class="dark:text-gray-300"> Vue.js</i><i class="dark:text-gray-300">, Vuex</i><i class="dark:text-gray-300">, Bootstrap Vue</i><i class="dark:text-gray-300">, Vuexy</i>
    </div>
    <div>
      <h3 class="font-bold text-lg">Warren Investimentos</h3>
      <p class="text-gray-700 font-semibold dark:text-white"> Desenvolvedor Frontend </p>
      <p class="dark:text-gray-300">Jan 2022 - Ago 2022</p><span class="dark:text-gray-300">Trabalhei com:</span><i class="dark:text-gray-300"> Vue.js</i><i class="dark:text-gray-300">, Vuex</i><i class="dark:text-gray-300">, Vuex.js</i><i class="dark:text-gray-300">, Typescript</i>
    </div>
    <div>
      <h3 class="font-bold">BoxTi</h3>
      <p class="text-gray-700 font-semibold dark:text-white">Desenvolvedor</p>
      <p class="dark:text-gray-300">Mar 2021- Dez 2021</p><span class="dark:text-gray-300">Trabalhei com:</span><i class="dark:text-gray-300"> HTML</i><i class="dark:text-gray-300">, CSS</i><i class="dark:text-gray-300">, Javascript</i><i class="dark:text-gray-300">, Vue.js</i><i class="dark:text-gray-300">, PHP</i><i class="dark:text-gray-300">, Node.js</i><i class="dark:text-gray-300">, MySQL</i>
    </div>
  </div>
</section>"
`;
